graph/telemetry/istio/appender/kiali_annotation.go

```diff
@@ -0,0 +1,69 @@
+package appender
+
+import (
+	"github.com/kiali/kiali/config"
+	"github.com/kiali/kiali/graph"
+	"github.com/kiali/kiali/models/annotation"
+)
+
+const AnnotationAppenderName = "kialiAnnotation"
+
+// Name: annotation
+type AnnotationAppender struct{}
+
+// Name implements Appender
+func (a AnnotationAppender) Name() string {
+	return AnnotationAppenderName
+}
+
+// AppendGraph implements Appender
+func (a AnnotationAppender) AppendGraph(trafficMap graph.TrafficMap, globalInfo *graph.AppenderGlobalInfo, namespaceInfo *graph.AppenderNamespaceInfo) {
+	if len(trafficMap) == 0 {
+		return
+	}
+
+	if getWorkloadList(namespaceInfo) == nil {
+		workloadList, err := globalInfo.Business.Workload.GetWorkloadList(namespaceInfo.Namespace)
+		graph.CheckError(err)
+		namespaceInfo.Vendor[workloadListKey] = &workloadList
+	}
+
+	a.applyAnnotationPresence(trafficMap, namespaceInfo)
+}
+
+func (a *AnnotationAppender) applyAnnotationPresence(trafficMap graph.TrafficMap, namespaceInfo *graph.AppenderNamespaceInfo) {
+	for _, n := range trafficMap {
+		// Skip the check if this node is outside the requested namespace, we limit badging to the requested namespaces
+		if n.Namespace != namespaceInfo.Namespace {
+			continue
+		}
```

Comment:
I think this check is valid as we onbly have the workloads for the current namespace.  Juts need to fix the comment.